src/test/java/com/endava/soa_5/cucumber/stepdef/SearchSteps.java

```diff
@@ -0,0 +1,64 @@
+package com.endava.soa_5.cucumber.stepdef;
+
+import com.endava.soa_5.actions.GenericActions;
+import com.endava.soa_5.actions.SearchPageActions;
+import com.endava.soa_5.context.ScenarioContext;
+import com.endava.soa_5.page_objects.SearchPage;
+import com.endava.soa_5.page_objects.enums.CategoryEnum;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.endava.soa_5.context.Keys.SEARCH_PAGE_CONFIRMATION;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SearchSteps extends GenericActions {
+    Logger log = LoggerFactory.getLogger(this.getClass());
+    SearchPageActions searchPageActions = new SearchPageActions();
+    SearchPage searchPage = new SearchPage();
+
+    @Given("^user insert (.+) into search field$")
+    public void userInsertProductIntoSearchField(String searchProduct) {
+        clickOnElement(searchPage.getSearchField());
+        searchPageActions.searchProduct(searchProduct);
+        log.info("---------------------User inserts keyword in search field--------------------");
```

Comment:
can log searchProduct and the input field to find easier  what testadata was used in case of failure. Only for username and password can be a general log, to not log sensitive data.

Reply:
Done!